kiwi/exceptions.py

```diff
@@ -795,3 +795,10 @@ class KiwiShellVariableValueError(KiwiError):
     Exception raised if a given python value cannot be converted
     into a string representation for use in shell scripts
     """
+
+
+class KiwiDecommissionedApiFunctionCalled(KiwiError):
```

Comment:
Maybe derive this from `DeprecationWarning` instead of `KiwiError`?

Reply:
I think we don't need an extra class that handles the deprecation in the KiwiError namespace. Actually those classes exists to handle errors of kiwi as an application. If used as an API the deprecation conditions should be handled on the python level. So I dropped the above on behalf of DeprecationWarning